.github/workflows/main.yml

```diff
@@ -499,3 +499,127 @@ jobs:
         with:
           name: 'obs-studio-macos-${{ matrix.arch }}-notarized'
           path: '${{ github.workspace }}/${{ env.FILE_NAME }}'
+
+  macos_sparkle:
+    name: '04 - macOS Sparkle Updates'
+    runs-on: [macos-12]
+    needs: [macos_release]
+    if: ${{ startsWith(github.ref, 'refs/tags/') && github.event_name != 'pull_request' }}
+    strategy:
+      matrix:
+        arch: ['x86_64', 'arm64']
+    env:
+      HAVE_SPARKLE_KEY: ${{ secrets.SPARKLE_PRIVATE_KEY != '' }}
+      SPARKLE_VERSION: '2.3.2'
+      SPARKLE_HASH: '2b3fe6918ca20a83729aad34f8f693a678b714a17d33b5f13ca2d25edfa7eed3'
+    defaults:
+      run:
+        shell: bash
+    steps:
+      - name: 'Checkout'
+        if: env.HAVE_SPARKLE_KEY == 'true'
+        uses: actions/checkout@v3
+        with:
+          path: 'repo'
+          ref: ${{ github.ref }}
+
+      - name: 'Download artifact'
+        if: env.HAVE_SPARKLE_KEY == 'true'
+        uses: actions/download-artifact@v3
+        with:
+          name: 'obs-studio-macos-${{ matrix.arch }}-notarized'
+          path: 'artifacts'
+
+      - name: 'Install Python requirements'
+        if: env.HAVE_SPARKLE_KEY == 'true'
+        run: pip3 install requests xmltodict
+
+      - name: 'Install Brew requirements'
+        if: env.HAVE_SPARKLE_KEY == 'true'
+        run: brew install coreutils pandoc
+
+      - name: 'Setup Sparkle'
+        if: env.HAVE_SPARKLE_KEY == 'true'
+        run: |
+          wget "https://github.com/sparkle-project/Sparkle/releases/download/${{ env.SPARKLE_VERSION }}/Sparkle-${{ env.SPARKLE_VERSION }}.tar.xz" -O Sparkle.tar.xz
+
+          if [[ '${{ env.SPARKLE_HASH }}' != "$(sha256sum Sparkle.tar.xz | cut -d " " -f 1)" ]]; then
+              echo "Sparkle download hash does not match!"
+              exit 1
+          fi
+
+          mkdir sparkle && cd sparkle
+          tar -xf ../Sparkle.tar.xz
+
+      - name: 'Setup folder structure'
+        if: env.HAVE_SPARKLE_KEY == 'true'
+        run: |
+          mkdir builds
+          mkdir -p output/appcasts
+          mkdir -p output/deltas/${{ matrix.arch }}
+
+      - name: 'Determine branch and tag'
+        if: env.HAVE_SPARKLE_KEY == 'true'
+        id: branch
+        run: |
+          cd repo/
```

Comment:
Make it a habit to use `pushd`/`popd` pairs to switch directory scopes instead of `cd` on CI (it won't make a difference here, but I'd like to remove any occurrence of `cd` calls moving forward).